reactor-core/src/main/java/reactor/core/publisher/FluxIterable.java

```diff
@@ -296,16 +299,17 @@ void slowPath(long n) {
 						return;
 					}
 
-					s.onNext(t);
-
-					if (cancelled) {
+					if (!b) {
+						s.onComplete();
+						onCloseWithDropError();
```

Comment:
I just noticed onClose is never actually provided. Maybe it's worth creating an issue to remove the implementation for it? It looks like the constructor accepting onClose is an internal one, so this wouldn't be a breaking change.

Reply:
it is used by FluxStream

Reply:
Oh. `FluxIterable#subscribe` with `onClose` is a static method used by `FluxStream`. I stand corrected, I only looked at constructors of `FluxIterable`.